packages/server/src/course/course.service.ts

```diff
@@ -101,4 +114,117 @@ export class CourseService {
       );
     }
   }
+
+  async registerCourses(
+    body: RegisterCourseParams[],
+    userId: number,
+  ): Promise<void> {
+    // obtains the ProfSectionGroupsModel of the professor
+    const profSectionGroups = await ProfSectionGroupsModel.findOne({
+      where: { profId: userId },
+    });
+
+    // iterate over each section group registration
+    for (const courseParams of body) {
+      // finds professor's section group with matching name
+      const sectionGroup = profSectionGroups.sectionGroups.find(
+        (sg) => sg.name === courseParams.sectionGroupName,
+      );
+      if (!sectionGroup)
+        throw new BadRequestException(
+          ERROR_MESSAGES.courseController.sectionGroupNotFound,
+        );
+      const khourySemesterParsed = this.loginCourseService.parseKhourySemester(
+        sectionGroup.semester,
+      );
+      const semester = await SemesterModel.findOne({
+        where: {
+          season: khourySemesterParsed.season,
+          year: khourySemesterParsed.year,
+        },
+      });
+      if (!semester)
+        throw new BadRequestException(
+          ERROR_MESSAGES.courseController.noSemesterFound,
+        );
+
+      // checks that course hasn't already been created
+      let course = await CourseModel.findOne({
+        where: {
+          name: courseParams.name,
```

Comment:
I don't think matching by the name will work, since the name is just any string entered by a prof - you could have two profs enter different display names. Matching by the sectionGroupName should work, though.

Reply:
Section group name in the same semester

Reply:
whoops, meant `sectionGroupName` here.